pkg/clustermanager/cluster_creator_test.go

```diff
@@ -0,0 +1,75 @@
+package clustermanager_test
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/pointer"
+
+	"github.com/aws/eks-anywhere/internal/test"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clustermanager"
+	"github.com/aws/eks-anywhere/pkg/clustermanager/mocks"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	mockswriter "github.com/aws/eks-anywhere/pkg/filewriter/mocks"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	mockskubeconfig "github.com/aws/eks-anywhere/pkg/kubeconfig/mocks"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+type clusterCreatorTest struct {
+	*WithT
+	ctx              context.Context
+	spec             *cluster.Spec
+	mgmtCluster      *types.Cluster
+	applier          *mocks.MockClusterApplier
+	writer           *mockswriter.MockFileWriter
+	kubeconfigWriter *mockskubeconfig.MockWriter
+}
+
+func newClusterCreator(t *testing.T) (*clustermanager.ClusterCreator, *clusterCreatorTest) {
+	ctrl := gomock.NewController(t)
+	cct := &clusterCreatorTest{
+		WithT:            NewWithT(t),
+		applier:          mocks.NewMockClusterApplier(ctrl),
+		writer:           mockswriter.NewMockFileWriter(ctrl),
+		kubeconfigWriter: mockskubeconfig.NewMockWriter(ctrl),
+		spec:             test.NewClusterSpec(),
+		ctx:              context.Background(),
+		mgmtCluster: &types.Cluster{
+			KubeconfigFile: "my-config",
+		},
+	}
+
+	cc := clustermanager.NewClusterCreator(cct.applier, cct.kubeconfigWriter, cct.writer)
+
+	return cc, cct
+}
+
+func (cct *clusterCreatorTest) expectFileCreate(clusterName, path string, w io.WriteCloser) {
+	cct.writer.EXPECT().Create(kubeconfig.FormatWorkloadClusterKubeconfigFilename(clusterName), gomock.AssignableToTypeOf([]filewriter.FileOptionsFunc{})).Return(w, path, nil)
+}
+
+func (cct *clusterCreatorTest) expectWriteKubeconfig(ctx context.Context, clusterName, kubeconfig string, w io.Writer) {
+	cct.kubeconfigWriter.EXPECT().WriteKubeconfig(ctx, clusterName, kubeconfig, w).Return(nil)
+}
+
+func (cct *clusterCreatorTest) expectApplierRun(ctx context.Context, spec *cluster.Spec, managementCluster types.Cluster) {
+	cct.applier.EXPECT().Run(ctx, spec, managementCluster).Return(nil)
+}
+
+func TestClusterCreatorCreateSync(t *testing.T) {
+	clusCreator, tt := newClusterCreator(t)
+	clusterName := tt.spec.Cluster.Name
+	path := "testpath"
+	writer := os.NewFile(uintptr(*pointer.Uint(0)), "test")
+	tt.expectApplierRun(tt.ctx, tt.spec, *tt.mgmtCluster)
+	tt.expectWriteKubeconfig(tt.ctx, clusterName, tt.mgmtCluster.KubeconfigFile, writer)
```

Comment:
nit: it's weird to pass members of the receiver as arguments to that receivers method